include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -304,12 +344,28 @@ class fm_index
                           | views::join(delimiter),
                           std::ranges::begin(tmp_text));
 
+        if (!reverse)
+        {
+            // we need at least one delimiter
+            if (number_of_texts == 1)
+                tmp_text.back() = delimiter;
 
-        // we need at least one delimiter
-        if (std::ranges::distance(text) == 1)
-            tmp_text.back() = delimiter;
-
-        std::ranges::reverse(tmp_text);
+            std::ranges::reverse(tmp_text);
+        }
+        else
+        {
+            // If only one text is in the text collection, we need at least one delimiter.
```

Comment:
This comment seems contradictory: What do we need to delimit, if there is just one text? And _at least one_ means there can be even multiple, or did you mean _at most_?
Maybe you can give some more details to avoid confusion here.

Reply:
Thanks!